components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/SecureFamily.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.studies.family.internal;
+
+import org.phenotips.data.Patient;
+import org.phenotips.data.internal.SecurePatient;
+import org.phenotips.studies.family.Family;
+import org.phenotips.studies.family.Pedigree;
+
+import org.xwiki.model.reference.DocumentReference;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONObject;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * A wrapper around PhenotipsFamily with no access to underlying XWikiDocument.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+public class SecureFamily implements Family
+{
+    private Family family;
+
+    /**
+     * Constructor that wraps around another (supposedly non-secure) instance of a Family.
+     *
+     * @param family the family to have a secure wrapper around
+     */
+    public SecureFamily(Family family)
+    {
+        if (family == null) {
+            throw new UnsupportedOperationException();
```

Comment:
It's either IllegalArgumentException or no exception at all like in all other constructors from this PR, like SecurePatient for example. For consistency I propose removing that check.